### PR TITLE
Search features by gene symbol and name

### PR DESCRIPTION
**Pull request by david4096**
Accompanies https://github.com/ga4gh/schemas/pull/639 and https://github.com/ga4gh/compliance/pull/196

In order to allow for some common access patterns to features data, it has been proposed to allow matching on gene name and gene symbol. This PR demonstrates that functionality within the reference server. 

The handling of requests has been improved for missing values and completed a TODO for generalizing query construction. This does not include a CLI interface, or modifications to the client.

### CODE REVIEW

File: ga4gh/datamodel/sequenceAnnotations.py [hunk 1]
```diff
@@ -76,35 +76,65 @@ def __init__(self, dbFile):
...
-            sql += ", ".join(["?", ] * len(featureTypes))
+            sql += ", ".join(["?", ] * len(kwargs.get('featureTypes')))
             sql += ") "
-            sql_args += tuple(featureTypes)
-        query = self._dbconn.execute(sql, sql_args)
-        return (query.fetchone())[0]
+            sql_args += tuple(kwargs.get('featureTypes'))
+        sql_rows += sql
+        sql_count += sql
+        return sql_rows, sql_args, sql_count
```
Comment (macieksmuga) [inline ga4gh/datamodel/sequenceAnnotations.py]: tiny quibble... wouldn't it be neater if the return list order were `sql_rows, sql_count, sql_args` (ie. string, string, list)

File: ga4gh/datamodel/sequenceAnnotations.py [hunk 2]
```diff
@@ -437,17 +463,19 @@ def getFeatures(self, referenceName, start, end,
...
                 start=start, end=end,
-                parentId=parentId, featureTypes=featureTypes)
+                parentId=parentId, featureTypes=featureTypes,
+                name=name, geneSymbol=geneSymbol)
 
         # pagination logic: None if last feature was returned,
         # else 1 + row number being returned (starting at row 0).
         if pageToken:
-            nextPageToken = int(pageToken)
+            nextPageToken = pb.int(pageToken)
```
Comment (macieksmuga) [inline ga4gh/datamodel/sequenceAnnotations.py]: Lead-encased integers? ;)

**Comment by macieksmuga** [comment]
+1 tested and works as intended

**Comment by kozbo** [comment]
+1

**Pull request merged**